fuel-core-interfaces/src/model/block.rs

```diff
@@ -16,14 +16,52 @@ use chrono::{
 };
 use core::ops::Deref;
 use fuel_vm::{
+    fuel_crypto,
     fuel_merkle,
     fuel_types::MessageId,
     prelude::Signature,
 };
+use std::fmt;
 
 #[cfg(any(test, feature = "test-helpers"))]
 use chrono::TimeZone;
 
+/// A cryptographically secure hash, identifying a block.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Default)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(transparent))]
+#[repr(transparent)]
+pub struct BlockId(Bytes32);
+
+impl BlockId {
+    /// Converts the hash into a message having the same bytes.
+    pub fn into_message(self) -> fuel_crypto::Message {
+        // This is safe because BlockId is a cryptographically secure hash.
+        unsafe { fuel_crypto::Message::from_bytes_unchecked(*self.0) }
+        // Without this, the signature would be using a hash of the id making it more
+        // difficult to verify.
+    }
+}
+
+impl From<Bytes32> for BlockId {
+    fn from(bytes: Bytes32) -> Self {
+        Self(bytes)
+    }
+}
+
+#[allow(clippy::from_over_into)] // Avoids circular dependency
+impl Into<Bytes32> for BlockId {
+    fn into(self) -> Bytes32 {
+        self.0
+    }
+}
```

Comment:
```suggestion
impl From<BlockId> for Bytes32 {
    fn from(block: BlockId) -> Self {
        block.0
    }
}
```
You should be able to still impl this as from here.

Reply:
replaced with derive_more